### DISCUSSION TITLE
Building compressed files with size limit from selection of files

### DISCUSSION THREAD

**Discussion by rcooknet** [aonez/Keka#1245] [Ideas]
Not sure if this has been suggested before, but my searching came up empty.

Quick use flow ... I need to upload ZIP files with a maximum file size ... let's say 50MB.  I want to select a group of files and build multiple independent ZIP files with a maximum single ZIP file size of 50MB.

I looked at the current Split functionality, but those do not appear to be independent ZIP files.

What I am looking for is a type of bin packing algorithm that optimally selects files to package together most efficiently to approach but not exceed a specified maximum compressed file size.  With the compression output size being an unknown during file selection, I realize this would be a tricky problem to solve most efficiently.

**Comment by rmzse** [comment]
Actually it exists, it is called "Split"https://github.com/aonez/Keka/wiki/Compressing-with-Keka#how-to-compress-some-files

**Reply by aonez** [reply]
No, he means independent ZIP files, not a ZIP file split in volumes.

**Comment by aonez** [comment]
@rcooknet what would be the use case for you? I see the big caveat when some files are bigger than the specified maximum size. Also this will be very hard to manage with (sub)folders and its contents, specially when extracting the contents of separate ZIP files all containing the same folder structure but different contents on it.

**Reply by rcooknet** [reply]
My use case is very simple compared to the issues you would have to address in a general feature.  I totally get that.

I need to upload ZIP files to a server for the sale of digital goods.  That server has a fixed limit for the files, let's say 50 MB.  So I have to manually break my digital assets into groups that package into ZIP files less than 50MB.

So something UI-wise similar to the current "Split" feature, but targeting one or more independent ZIP files.  Sub-folders add a whole level of complexity that I don't have in my use case. Optimal binning of files isn't a top priority for me ... close enough works.

**Reply by aonez** [reply]
I see.

If you can bulk upload those ZIP files you can try the "Archive items separately" option so Keka will create a single ZIP file for every dropped file/folder. Maybe that works for you.
